### PR TITLE
Adding Pumpkin SDK target

### PR DESCRIPTION
**Pull request by CatGarab**
**PR Overview**:

Several targets have been added/modified
  - kubos-linux-beaglebone-gcc - New target for KubOS Linux on a regular Beaglebone Black
  - legacy-linux-beaglebone-gcc - Renamed. Used to be kubos-linux-beaglebone-gcc. This is for creating SDK projects to run on a BBB that's not running our Linux.
  - kubos-linux-pumpkin-mbm2-gcc - New target for Pumpkin MBM2 board
  - kubos-linux-gcc - New parent target for all KubOS Linux boards. Contains the flash script and common configuration settings, since all of them should be the same.

Note: There is no actual difference between kubos-linux-beaglebone-gcc and kubos-linux-pumpkin-mbm2-gcc; since they use the same processor, they have the same toolchain. However, I think it'll be less confusing for users if there are two separate targets.

**Documentation**:

- Updated targets documentation
- Added new docs for Pumpkin MBM2 board

**Integration Tests**:

Tests have not been created yet. They will be done in conjunction with [this card](https://trello.com/c/LijUOJSh/12-pumpkin-qa-suite)

### CODE REVIEW

**Comment by CatGarab** [comment]
[Pumpkin SDK target](https://trello.com/c/6QsOOpPC/23-pumpkin-sdk-target)

**Comment by CatGarab** [comment]
[KubOS Linux docs updates for pumpkin](https://trello.com/c/2KD8GDBx/15-kubos-linux-docs-updates-for-pumpkin)

File: docs/kubos-linux-on-mbm2.rst [hunk 1]
```diff
@@ -0,0 +1,480 @@
+KubOS Linux on the Pumpkin MBM2
```
Comment (davesims) [inline docs/kubos-linux-on-mbm2.rst]: A small 😂 of joy for all this awesomeness @CatGarab :point_down:. Just beautiful. 

❤️ ✖️ :100: ✖️ 💯

File: docs/kubos-linux-on-mbm2.rst [hunk 2]
```diff
@@ -0,0 +1,480 @@
...
+.. note:: Just looking to install KubOS Linux onto a Pumpkin MBM2? Skip ahead to the :ref:`installation-process-mbm2`.
+
+The goal of this document is to create a KubOS Linux installation for the Pumpkin
+Motherboard Module 2 that can then run the satellite services (telemetry, payload 
+communication, etc) needed for the Pumpkin customers' missions.
+
+The :doc:`Working with the Pumpkin MBM2 <working-with-the-mbm2>` doc can then be used to
+create and load a user application on top of the new KubOS Linux install.
+
+Ideally, the user should never have to mess with the kernel themselves.
```
Comment (davesims) [inline docs/kubos-linux-on-mbm2.rst]: Grammar nit: ~~themselves~~.

File: docs/kubos-linux-on-mbm2.rst [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+U-Boot
+~~~~~~
+
+`Wiki <https://en.wikipedia.org/wiki/Das_U-Boot>`__
+
+`Site Page <http://www.denx.de/wiki/U-Boot>`__
+
+`Kubos U-Boot Repo <https://github.com/kubostech/uboot>`__
+
+U-Boot, at the most basic level, is responsible for loading the kernel from the
```
Comment (davesims) [inline docs/kubos-linux-on-mbm2.rst]: 😍 Great technical writing @CatGarab. I'm geeking out.

File: docs/kubos-linux-on-mbm2.rst [hunk 4]
```diff
@@ -0,0 +1,480 @@
...
+
+    $ sudo make BR2_EXTERNAL=../kubos-linux-build pumpkin-mbm2_defconfig
+
+Build everything
+
+::
+
+    $ sudo make
+
+The full build process will take a while. Running on a Linux VM, it took about
```
Comment (davesims) [inline docs/kubos-linux-on-mbm2.rst]: Nit: s/took/takes/

File: docs/kubos-linux-on-mbm2.rst [hunk 5]
```diff
@@ -0,0 +1,480 @@
...
+    $ reset
+    
+These commands will:
+
+  - Restore the relevant environment variables to their default values
+  - Save the new values to persistent storage
+  - Reboot the system
+  
+As long as a valid kernel and rootfs are available, your system should now successfully boot
+into KubOS Linux.
```
Comment (davesims) [inline docs/kubos-linux-on-mbm2.rst]: @CatGarab I can't say enough about the thoroughness and solid technical writing in all of this :point_up:. 

👏 👏 👏

**Pull request merged**